warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -1897,14 +1897,70 @@ public static final GeoTimeSerie subSerie(GeoTimeSerie gts, long starttimestamp,
     // Extract values/locations/elevations that lie in the requested interval
     //
 
-    // We know how many data will the new GTS so we provision arrays to receive the data.
+    // We know how many data the new GTS will have so we provision arrays to receive the data.
     int count = lastidx - firstidx + 1;
+
     GTSHelper.multiProvision(subgts, gts.type, count, count);
 
-    for (int i = firstidx; i <= lastidx; i++) {
-      setValue(subgts, gts.ticks[i], null != gts.locations ? gts.locations[i] : GeoTimeSerie.NO_LOCATION, null != gts.elevations ? gts.elevations[i] : GeoTimeSerie.NO_ELEVATION, valueAtIndex(gts, i), overwrite);
-    }
+    if (!overwrite) {
+      if (count > 0) {
+        if (null != gts.locations) {
+          if (null != subgts.locations && subgts.locations.length >= count) {
+            System.arraycopy(gts.locations, firstidx, subgts.locations, 0, count);
+          } else {
+            subgts.locations = Arrays.copyOf(gts.locations, count);
+          }
+        } else {
+          subgts.locations = null;
+        }
+        if (null != gts.elevations) {
+          if (null != subgts.elevations && subgts.elevations.length >= count) {
+            System.arraycopy(gts.elevations, firstidx, subgts.elevations, 0, count);
+          } else {
+            subgts.elevations = Arrays.copyOf(gts.elevations, count);
+          }
+        } else {
+          subgts.elevations = null;
+        }
+        if (null != subgts.ticks) {
+          System.arraycopy(gts.ticks, firstidx, subgts.ticks, 0, count);
+        }
 
+        switch(gts.type) {
+          case LONG:
+            if (null != subgts.longValues) {
```

Comment:
We don't really need this check since multiProvision has created the array since count > 0.
However, we can do differently by doing the same thing than for geo arrays (and not use multiProvision in the case !overwrite). That would save writing twice in the same array when we expand and copy.